src/realm/table.cpp

```diff
@@ -1627,6 +1643,7 @@ ref_type Table::create_empty_table(Allocator& alloc, TableKey key)
     top.add(0); // Sequence number
     top.add(0); // Collision_map
     top.add(0); // pk col key
+    top.add(0); // flags
 
     REALM_ASSERT(top.size() == top_array_size);
```

Comment:
I guess `flags` is an optional field in file format 10 realm files. So we cannot assume it is present.

Reply:
I still have the opinion that the field should be optional. Although we will bump the file format version to 11, we should still be able to open version 10 files in read-only mode.